add support for progressive JPEG emulation

gweedo767 opened the pull request:
This adds support for Progressive JPEG emulation to add even more terribleness.  The only real issue with this setup is we can't pull the images via getElementsByTagName() since that gets a live collection that would grow infinitely each time a new blurred image is created.  I used querySelectorAll('img:not([class="progessiveJPEGemulator"])') instead only when the progressiveJPEG parameter is set to true.

gweedo767 committed:
add support for progressive JPEG emulation

kand commented:
Looks cool! I'll take it for a test run then merge it in if it's all good.

gweedo767 commented:
querySelectorAll() could honestly be used in both cases. If the library was updated in the future to live apply this affect to newly generated img's it would matter again.

gweedo767 commented:
Any thoughts on this?  I can update it to use querySelectorAll() in all cases as well if wanted since the fact that getElementsByTagName() doesn't help in any use case here.

kand commented:
Yea let's go the route of just using querySelectorAll(), seems like the best route here.

Honestly, I'm not super thrilled with making new imgs on the page, but as far as I can tell this seems to be the easiest way to achieve this blur effect.

If you make that change to the elements argument I'll merge in this request.

gweedo767 committed:
use querySelectorAll() in all cases

since we aren’t using the HTML LiveCollection returned by
getElementsByTagName() there is no reason to not just use
querySelectorAll() in all cases.

gweedo767 commented:
Updated, let me know if you have any other things you want to see modified in this.

kand commented:
Done, thanks.
